### PR TITLE
Added the ability to generate signals that start and/or end outside o…

### PR DESCRIPTION
**Pull request by wcheadley**
…f the observation interval.

### CODE REVIEW

File: pywaspgen/burst_datagen.py [hunk 1]
```diff
@@ -94,7 +108,7 @@ def gen_burstlist(self, burst_idx=-1):
                 burst_list.append(candidate_burst)
                 sig_count += 1
             elif not self.config["spectrum"]["allow_collisions_flag"]:
-                if not self.__check_collisions(candidate_burst, burst_list):
+                if not self.__check_collisions(candidate_burst, burst_list) and not self.__check_not_observed(candidate_burst):
```
Comment (cmmcirvin) [inline pywaspgen/burst_datagen.py]: This logic will only prevent the burst from being added when the collisions flag is off. This check needs to go around all cases.

File: configs/default.json [hunk 2]
```diff
@@ -20,16 +20,16 @@
       { "format": "qam", "order": 16, "label": "16QAM" },
       { "format": "qam", "order": 64, "label": "64QAM" }
     ],
-    "max_signals": 10,
+    "max_signals": 100,
```
Comment (cmmcirvin) [inline configs/default.json]: Is this change intentional? Looking at the outputs generated by this config file, only ~20 signals are actually generated in each burst list due to collisions.

**Comment by wcheadley** [comment]
Caleb found a few logic errors and fixed them and I further fixed a slight issue with the plotting when signals start or end outside the observation interval.

**Review by cmmcirvin — APPROVED** [review]
